### ISSUE TITLE
Strange error when writing merged streams to SEGY

### ISSUE THREAD

**Issue by LKueperkoch** [obspy/obspy#1385]
Hello ObsPy-Com,

I am struggling with converting a stream (read from SEG2-data files, created with SUMMIT-VIPA) to a SEGY-file. Attached you find the code. The script searches for recent files in a given directory, which will be filled with SEG2 data via FTP server. As soon as there is a certain number (numdatpack) of data files per station available, the data is read in and merged to one stream object, as for further processing all data (of all stations) must be stored within one data file (beside this, there are some smaller actions like moving, renaming the data files and converting them to miniSEED). Furthermore, because only 1-minute data files are available, the traces are merged to longer traces. The data is visualized correctly (including data gaps), but converting to SEGY with obspy.core.read fails (no write support for SEG2), a strange error saying "error: short format requires SHRT_MIN <= number <= SHRT_MAX" occurs, both on Windows and Linux system. I am using Windows 7, Ubuntu 14.04, the latest ObsPy version and Python 2.7. Furthermore I attached some SEG2-data files. Does anyone have any idea?
Thanks in advance

Ludger

[WatchFTPData.txt](https://github.com/obspy/obspy/files/238574/WatchFTPData.txt)
[SEG2_data.tar.gz](https://github.com/obspy/obspy/files/238577/SEG2_data.tar.gz)

**Comment by megies** [comment]
Hi Ludger,
can you break down the code to a minimal example? That would make it much easier for people to start working on this issue. I think the issue should manifest in 2-3 lines (read data file that causes the issue, write to SEGY). That's is a really long code piece you posted there..

**Comment by LKueperkoch** [comment]
Hi Tobi,

attached you find the main part of the code, still pretty long, but the error (if there is one) might be somewhere there ...
[WatchFTPData_snipped.txt](https://github.com/obspy/obspy/files/238807/WatchFTPData_snipped.txt)

**Comment by LKueperkoch** [comment]
O.k., I'd tried to break it down a little bit more ....
[WatchFTPData_snipped.txt](https://github.com/obspy/obspy/files/238819/WatchFTPData_snipped.txt)

**Comment by LKueperkoch** [comment]
This issue is still open, forgive me! By the way: writing to miniSEED is working properly. But if I reload the (merged) miniSEED data and want to write it as SEGY, the error again occurs ....

Ludger

**Comment by LKueperkoch** [comment]
Maybe this is now a little bit more clearly. 
[WatchFTPData_snipped.txt](https://github.com/obspy/obspy/files/238861/WatchFTPData_snipped.txt)

**Comment by megies** [comment]
This is still endlessly long with nested try/excepts and whatnot. Can you just post the single data file you wrote to miniseed and that can't be written to SEGY?

**Comment by LKueperkoch** [comment]
I tried to break it down even more. However, because the bug must be somewhere within the (nested) loops for merging the streams and traces, I can not shorten it more, as it then is no more clear what's going on. However, I think the code is not that complex, it is straight forward. Furthermore, it is not a thing of the miniseed file, I am able to write to miniseed but (with the same streams and traces) not to SEGY! Thus, there might be something strange in the header, which might has been wrongly adopted within the merging actions. Maybe a hint: if I choose stream.plot(), the figure is shaded red indicating overlaps ...
The SEG2 data to be read and merge are already available within my initial comment.
Sorry for bothering you with my quick and dirty code!
[WatchFTPData_snipped.txt](https://github.com/obspy/obspy/files/239988/WatchFTPData_snipped.txt)

**Comment by LKueperkoch** [comment]
Solved! It is not a bug, the code is o.k., I only had to implement a stream.trim in line 55. Then it works. Thus, I will close this issue. However, the next problem occurs: our online-processing system is not able to handle this SEGY format, saying "SEGY data format not supported". Great! Maybe this will be the next issue ....
Thanks,

Ludger
[WatchFTPData_snipped.txt](https://github.com/obspy/obspy/files/241869/WatchFTPData_snipped.txt)
